### PR TITLE
tee-supplicant: add TEE_RPC_LOAD_TA2

### PR DESCRIPTION
**Pull request by jenswi-linaro**
Adds new RPC function TEE_RPC_LOAD_TA2 which doesn't allocate a buffer
for the TA to load, instead secure world is responsible to allocate the
buffer when doing this RPC.

Tested-by: Jens Wiklander <jens.wiklander@linaro.org> (QEMU)
Signed-off-by: Jens Wiklander <jens.wiklander@linaro.org>

### CODE REVIEW

File: tee-supplicant/src/tee_supplicant.c [hunk 1]
```diff
@@ -358,6 +358,58 @@ static void free_ta(struct tee_rpc_invoke *inv)
 	OUTMSG();
 }
 
+static void load_ta2(int fd, struct tee_rpc_invoke *inv)
+{
+	void *ta = NULL;
+	int ta_found = 0;
```
Comment (jforissier) [inline tee-supplicant/src/tee_supplicant.c]: Initialization not needed for `ta_found`.

File: tee-supplicant/src/tee_supplicant.c [hunk 2]
```diff
@@ -358,6 +358,58 @@ static void free_ta(struct tee_rpc_invoke *inv)
...
+	}
+
+	if (get_param(fd, inv, 0, &shared_mem)) {
+		inv->res = TEEC_ERROR_BAD_PARAMETERS;
+		return;
+	}
+	cmd = (struct tee_rpc_ta *)shared_mem.buffer;
+
+	ta_found = TEECI_LoadSecureModule(devname1, &cmd->uuid, &ta, &size);
+	/* Tracked by 6408 */
```
Comment (jforissier) [inline tee-supplicant/src/tee_supplicant.c]: What does this mean? -> remove

File: tee-supplicant/src/tee_supplicant.c (hunk 1 above)
Comment (jenswi-linaro) [inline tee-supplicant/src/tee_supplicant.c]: OK, I'll fix.

File: tee-supplicant/src/tee_supplicant.c (hunk 2 above)
Comment (jenswi-linaro) [inline tee-supplicant/src/tee_supplicant.c]: OK, I'll fix.

File: tee-supplicant/src/tee_supplicant.c [hunk 3]
```diff
@@ -358,6 +358,58 @@ static void free_ta(struct tee_rpc_invoke *inv)
...
+	int ta_found = 0;
+	size_t size = 0;
+	struct tee_rpc_ta *cmd;
+	TEEC_SharedMemory shared_mem;
+
+	INMSG();
+
+	if (inv->nbr_bf < 2) {
+		inv->res = TEEC_ERROR_BAD_PARAMETERS;
+		return;
```
Comment (pascal-brand-st-dev) [inline tee-supplicant/src/tee_supplicant.c]: Missing `OUTMSG`

File: tee-supplicant/src/tee_supplicant.c [hunk 4]
```diff
@@ -358,6 +358,58 @@ static void free_ta(struct tee_rpc_invoke *inv)
...
+	}
+
+	if (get_param(fd, inv, 0, &shared_mem)) {
+		inv->res = TEEC_ERROR_BAD_PARAMETERS;
+		return;
```
Comment (pascal-brand-st-dev) [inline tee-supplicant/src/tee_supplicant.c]: Missing `OUTMSG`

**Comment by pascal-brand-st-dev** [comment]
With my comments:

    Reviewed-by: Pascal Brand <pascal.brand@linaro.org>

**Comment by jforissier** [comment]
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`
`Tested-by: Jerome Forissier <jerome.forissier@linaro.org> (HiKey)`

**Comment by jenswi-linaro** [comment]
Comments addressed, tags applied.

**Pull request merged**